### PR TITLE
Abi Unit and Integration tests for getActionItem routes and Controllers

### PR DESCRIPTION
**Pull request by Abi-Liu**
# Description
Implementing unit and integration tests for the `actionItemController` and `actionItemRoutes`. These tests cover the post and get methods for the routes/controllers

## Related PRS (if any):
N/A
…

## Main changes explained:
- Implements unit tests for the getactionItem and postactionItem method in the getActionController.
- Implements integration tests for the get and post routes in `actionItemRouter`
- Tests the positive and negative and applicable edge case for all test cases
…

## How to test:
1. check into current branch
2. do `npm install` 
3. run `npm run test actionItem`
4. verify that all tests pass.

## Screenshots or videos of changes:

Screenshot 2024-03-27 at 11 43 11 AM

### CODE REVIEW

**Comment by DiegoSalas27** [comment]
Hi Abi,

Very nice job. All of your tests are passing with the correct values. Tests are meaningful and cover both positive and edge cases.

image

Tomorrow we will merge this PR: https://github.com/OneCommunityGlobal/HGNRest/pull/813

That PR contains integration tests for the postUserProfile method I unit tested, and it will be a perfect template for you to use and finish both of your PRs.

Very happy to have you on the team!

**Review by DiegoSalas27 — APPROVED** [review]

File: src/test/db/createUser.js [hunk 1]
```diff
@@ -46,9 +46,17 @@ const createUser = async () => {
   up.isFirstTimelog = true;
   up.actualEmail = '';
   up.isVisible = true;
-  up._id = '65cf6c3706d8ac105827bb2e';
 
-  await up.save();
+  /*
```
Comment (DiegoSalas27) [inline src/test/db/createUser.js:50]: This is fine. This update will conflict with my PR, but I will my PR accordingly. to make sure it doesn't break your tests. Alternatively, I will create another file just to hardoce the _id for testing purposes.

**Review by DiegoSalas27 — CHANGES_REQUESTED** [review]
Very nice work, Abi. I have tested your PR and it does cover every positive and negative cases for all of the methods od the actionItemController. The integration tests were smart and made sure you postActionItem and getActionItem functions work correctly to support their API route.

We need to add the integration test for deleteActionItem and editActionItem to make sure those functions work properly.

Please add those missing unit tests, and I will merge your PR.

I am very happy for the hard work, you are doing a great job. Please add the rest of our team mates as reviewers for them to have a look at how to properly unit test the backend.

File: src/test/db/createUser.js (hunk 1 above)
Comment (Abi-Liu) [reply src/test/db/createUser.js:50]: I see, I thought it would just be more reusable to have it create new _id's in the event that we need to create multiple users to test functionality.

Comment (DiegoSalas27) [reply src/test/db/createUser.js:50]: That makes sense. I will make sure this file is untouched, and will create a new one for my own unit tests.

**Review by linkhanthtel — APPROVED** [review]
Hi Abi,

I reviewed your PR and all tests have passed!

**Comment by Gayathr-2012** [comment]
Hi Abi,

I reviewed your PR and all tests have passed!


image

image

**Review by DiegoSalas27 — APPROVED** [review]
Hi Abi,

Your PR's unit tests and integration tests are bulletproof. They cover everything that's needed to make a robust backend: positive, negative and edge cases are accounted for in both unit tests and integration tests.

This is a master work. Congratulations, and thank you!

https://github.com/OneCommunityGlobal/HGNRest/assets/26840798/10ee6fac-f20e-4bc6-91fa-9a0cd4810f84
